homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -57,6 +57,10 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
     coordinator = data[COORDINATOR]
     name = data[NAME]
 
+    # Add a Unique ID to previously discovered entities that were not given an ID
```

Comment:
We should update the config entry from or in `async_setup_entry` so the missing id will be there

Example:

https://github.com/home-assistant/core/blob/dev/homeassistant/components/yeelight/__init__.py#L263